version.sh

```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+if [ -n "$PROJECT_VERSION" ]; then
+    # use env variable PROJECT_VERSION if defined
+    echo $PROJECT_VERSION
```

Comment:
This looks should do for the use-case I was concerned about: building docker images based on a development source folder without including all other branches, reflogs, stashed patches and configuration such as remote repos into the image.

Does this run during the Maven build? If so, we can basically run the below git command _outside_ of the build container, and pass it as a `--build-arg`.